drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/models/ScenarioGridModel.java

```diff
@@ -313,25 +322,14 @@ public void updateHeader(int columnIndex, int rowIndex, String value) {
         simulation.getSimulationDescriptor().getFactMappingByIndex(columnIndex).setExpressionAlias(value);
     }
 
-    public Range setNewCellValue(int rowIndex, int columnIndex, GridCellValue<?> value) {
-        return setNewCell(rowIndex, columnIndex, () -> new ScenarioGridCell((GridCellValue<String>) value));
-    }
-
-    public Range deleteNewCell(int rowIndex, int columnIndex) {
-        FactMapping factMapping = simulation.getSimulationDescriptor().getFactMappingByIndex(columnIndex);
-        simulation.getScenarioByIndex(rowIndex)
-                .removeFactMappingValueByIdentifiers(factMapping.getFactIdentifier(), factMapping.getExpressionIdentifier());
-        return super.deleteCell(rowIndex, columnIndex);
-    }
-
     public void clear() {
         // Deleting rows
         int to = getRowCount();
         IntStream.range(0, to)
                 .map(i -> to - i - 1)
-                .forEach(this::deleteRow);
+                .forEach(super::deleteRow);
```

Comment:
If the reverse order is really so important, I think we could use the code below, let me know what do you think.
```java
IntStream.range(0, to - 1)
                .boxed()
                .sorted(Collections.reverseOrder())
                .forEach(super::deleteRow);
```

Reply:
The scope of that code was actually to avoid boxing and reverting order - with that implementation I loop just once; with your's, two (beside the boxing)

Reply:
ok